torch/nn/modules/instancenorm.py

```diff
@@ -36,29 +36,90 @@ def eval(self):
         return self
 
 
+class LayerNorm(_InstanceNorm):
+    r"""Applies Layer Normalization over a 2D input that is seen
+    as a mini-batch of 1D inputs.
+
+    .. math::
+
+        y = \gamma * \frac{x - \mu_x}{\sigma_x + \epsilon} + \beta
+
+    The mean and standard deviation are calculated per-dimension separately
+    for each object in a mini-batch. Gamma and beta are learnable parameter
+    vectors of size C (where C is the input size).
+
+    During training, this layer keeps a running estimate of its computed mean
+    and variance. The running sum is kept with a default momentum of 0.1.
+
+    At evaluation time (`.eval()`), the default behaviour of the LayerNorm
+    module stays the same, i.e. the running mean/variance is NOT used for
+    normalization. One can force using the stored mean and variance with
+    the `.train(False)` method.
```

Comment:
How does this work? As far as I can tell, `eval()` just calls `train(False)`. I see that `_InstanceNorm` overrides `eval()`, but:

1) eval() isn't called recursively. If you call `eval()` on a container, it'll call `train(False)` on it's children.
2) `_InstanceNorm` doesn't appear to use `self.training` so I don't see how you can "force" using mean/var.